DUI3-DX/Converters/ArcGIS/Speckle.Converters.ArcGIS3/ArcGISConversionContextStack.cs

```diff
@@ -1,18 +1,99 @@
 using System.Diagnostics.CodeAnalysis;
-using ArcGIS.Core.Geometry;
+using ArcGIS.Core.Data.DDL;
+using ArcGIS.Core.Data;
+using ArcGIS.Desktop.Core;
+using ArcGIS.Desktop.Framework.Threading.Tasks;
 using ArcGIS.Desktop.Mapping;
 using Speckle.Converters.Common;
 
 namespace Speckle.Converters.ArcGIS3;
 
+public record ArcGISDocument(Project Project, Map Map, Uri SpeckleDatabasePath);
+
 // POC: Suppressed naming warning for now, but we should evaluate if we should follow this or disable it.
 [SuppressMessage(
   "Naming",
   "CA1711:Identifiers should not have incorrect suffix",
   Justification = "Name ends in Stack but it is in fact a Stack, just not inheriting from `System.Collections.Stack`"
 )]
-public class ArcGISConversionContextStack : ConversionContextStack<Map, Unit>
+public class ArcGISConversionContextStack : ConversionContextStack<ArcGISDocument, ACG.Unit>
 {
-  public ArcGISConversionContextStack(IHostToSpeckleUnitConverter<Unit> unitConverter)
-    : base(MapView.Active.Map, MapView.Active.Map.SpatialReference.Unit, unitConverter) { }
+  public ArcGISConversionContextStack(IHostToSpeckleUnitConverter<ACG.Unit> unitConverter)
+    : base(
+      new ArcGISDocument(Project.Current, MapView.Active.Map, EnsureOrAddSpeckleDatabase()),
+      MapView.Active.Map.SpatialReference.Unit,
+      unitConverter
+    ) { }
+
+  private const string FGDB_NAME = "Speckle.gdb";
+
+  public static Uri EnsureOrAddSpeckleDatabase()
+  {
+    return AddDatabaseToProject(GetDatabasePath());
+  }
+
+  public static Uri GetDatabasePath()
```

Comment:
1. Why we have it static? Can't we retrieve from instance?
2. Why we catch specific exception types and rethrow regardless?